demos/mri_reconstruction_demo/cpp/main.cpp

```diff
@@ -0,0 +1,167 @@
+// Copyright (C) 2018-2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+#include <inference_engine.hpp>
+#include <opencv2/opencv.hpp>
+
+#include "mri_reconstruction_demo.hpp"
+#include "npy_reader.hpp"
+
+using namespace InferenceEngine;
+
+bool ParseAndCheckCommandLine(int argc, char *argv[]);
+
+static cv::Mat infEngineBlobToMat(const Blob::Ptr& blob);
+
+struct MRIData {
+    cv::Mat data;
+    cv::Mat samplingMask;
+    cv::Mat reconstructed;
+    // Hybrid-CS-Model-MRI/Data/stats_fs_unet_norm_20.npy
+    std::vector<double> stats = {2.20295299e-1, 1.11048916e+3, 4.16997984, 4.71741395};
+};
+
+static const std::string kWinName = "MRI reconstruction with OpenVINO";
+
+void callback(int pos, void* userdata);
+
+cv::Mat kspaceToImage(const cv::Mat& kspace);
+
+float psnr(const cv::Mat& data0, const cv::Mat& data1, float maxVal);
+
+int main(int argc, char** argv) {
+    // ------------------------------ Parsing and validation of input args ---------------------------------
+    if (!ParseAndCheckCommandLine(argc, argv)) {
+        return 0;
+    }
+
+    Core ie;
+
+    CNNNetwork net = ie.ReadNetwork(FLAGS_m);
+    net.getInputsInfo().begin()->second->setLayout(Layout::NHWC);
+
+    ExecutableNetwork execNet = ie.LoadNetwork(net, FLAGS_d);
+    InferRequest infReq = execNet.CreateInferRequest();
+
+    // Hybrid-CS-Model-MRI/Data/sampling_mask_20perc.npy
+    MRIData mri;
+    mri.samplingMask = blobFromNPY(FLAGS_p);
+    std::cout << "Sampling ratio: " << 1.0 - cv::mean(mri.samplingMask)[0] << std::endl;
+
+    mri.data = blobFromNPY(FLAGS_i);
+    CV_Assert(mri.data.depth() == CV_64F);
+    const int numSlices = mri.data.size[0];
+    const int height = mri.data.size[1];
+    const int width = mri.data.size[2];
+    mri.data /= sqrt(height * width);
+
+    mri.reconstructed.create({numSlices, height, width}, CV_8U);
+
+    std::cout << "Compute..." << std::endl;
+
+    cv::Mat inputBlob = infEngineBlobToMat(infReq.GetBlob(net.getInputsInfo().begin()->first));
+    cv::Mat outputBlob = infEngineBlobToMat(infReq.GetBlob(net.getOutputsInfo().begin()->first));
+    outputBlob = outputBlob.reshape(1, height);
+
+    cv::TickMeter tm;
+    tm.start();
+    for (int i = 0; i < numSlices; ++i) {
+        // Prepare input
+        cv::Mat kspace = cv::Mat(height, width, CV_64FC2, mri.data.ptr<double>(i)).clone();
+
+        kspace.setTo(0, mri.samplingMask);
+        kspace = (kspace - mri.stats[0]) / mri.stats[1];
```

Comment:
```suggestion
        kspace = (kspace - cv::Scalar(mri.stats[0], mri.stats[0])) / mri.stats[1];
```
There is a warning `[ WARN:0] global C:\jenkins\workspace\OpenCV\OpenVINO\2021.4\build\windows\opencv\modules\core\src\matrix_expressions.cpp (1334) cv::MatOp_AddEx::assign OpenCV/MatExpr: processing of multi-channel arrays might be changed in the future: https://github.com/opencv/opencv/issues/16739`